### DISCUSSION TITLE
Save canceled_at in Pay::Subscription

### DISCUSSION THREAD

**Discussion by fschwahn** [pay-rails/pay#883] [Q&A]
We'd like to show in our backend when a subscription has been canceled, but has not ended yet. That's possible using the `canceled`-scope. 

However, we'd also like to show recent cancelations - ie. order subscriptions by cancelation date, to easily see the customers who most recently cancelled their contract.

Would you consider adding support that?

**Comment by excid3** [comment]
A cancellation can happen immediately or at the end of the period, so you could extend the class with before_save callback that saves to a column (or json attribute) you add to the Pay::Subscription model. You'd need to make sure it handles cancel now and cancel in the future situations.

I'm undecided if we want to add that to the gem or not. It seems more fitting for event tracking than a core piece of the payment code.
